### PR TITLE
Add diff and check mode to cron module

### PR DESCRIPTION
**Pull request by towolf**
##### Issue Type:
- Feature Pull Request
##### Plugin Name:

cron
##### Summary:

The cron module was missing both check and diff modes. 

Diff mode is a major new feature on the 2.1 Roadmap.
##### Example:

example screenshot

### CODE REVIEW

**Comment by towolf** [comment]
Notice that there are two logic errors in the cron module.

First, the default user is stated to be `root`. This is not reflected in the code. The default user is whoever the task is run as. Not setting `user` leaves it as `None`. (Only left this in the screenshot for illustration. I removed the spurious output in the final PR.) (docs issue)

The second is that a cron.d file is only unlinked when `name` is omitted. Removing the last named item in a cron.d snippet with leave behind an empty file. (docs issue)

**Comment by resmo** [comment]
Thanks @towolf. To the current maintainers, @dsummersl, @EvanK please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

File: system/cron.py [hunk 1]
```diff
@@ -593,14 +604,22 @@ def main():
         special_time = "reboot"
 
     # if requested make a backup before making a change
-    if backup:
+    if backup and not module.check_mode:
```
Comment (dsummersl) [inline system/cron.py:607]: nice catch...

**Comment by EvanK** [comment]
shipit

**Comment by resmo** [comment]
Thanks again to @towolf for this PR, and thanks @dsummersl, @EvanK for reviewing. Marking for inclusion.

**Pull request merged**